### DISCUSSION TITLE
RSpec Workflow Fix (Credentials)

### DISCUSSION THREAD

**Discussion by patricklsamson** [avionschool/batch8-activities#24] [General]
Failure in RSpec Workflow Test in pull request before merging

![image](https://user-images.githubusercontent.com/79001601/129467882-e1ff53fb-11a1-4f81-acba-fbeb5d421421.png)
![image](https://user-images.githubusercontent.com/79001601/129467885-31efe26a-fd12-4e1d-8e78-7bb4c173b39b.png)

Fix:

From root of Rails Application, delete the following
./config/credentials.yml.enc

In terminal, go to the root of Rails Application then enter the following:

![image](https://user-images.githubusercontent.com/79001601/129467897-536542b8-9c44-458b-b6c5-8c5ff814d31c.png)

Copy and save the secret_key_base generated inside the new credentials.yml.enc temporarily then exit to save. New credentials.yml.enc and master.key will be created.

Back again to the terminal enter the following:

![image](https://user-images.githubusercontent.com/79001601/129467949-ad5ff815-1f04-40b4-891a-dc3e8b60c1f7.png)

Paste inside the secret_key_base. Save and exit.

Edit the GitHub Workflow for RSpec. From the root of the Rails Application, go to: ./.github/workflows/rspec.yml

Go to the bottom of the rspec.yml change "RAILS_MASTER_KEY: ${{ secrets.RAILS_MASTER_KEY }}" to "RAILS_TEST_KEY: ${{ secrets.RAILS_TEST_KEY }}"

![image](https://user-images.githubusercontent.com/79001601/129468024-5f2f527f-3f52-4c21-8aa1-f3dfbb3a6656.png)

Sources:
https://stackoverflow.com/questions/66273852/rails-credential-values-are-nil-while-running-tests-on-github-actions
https://stackoverflow.com/questions/67172951/rails-github-actions-credentials
https://medium.com/craft-academy/encrypted-credentials-in-ruby-on-rails-9db1f36d8570

**Comment by lea-tinoso** [comment]
Hi, just an additional info on this:

When @patricklsamson  said secret_key_base he actually means the contents of credential.yml.enc file.

Also you might/might not encounter this issue
![image](https://user-images.githubusercontent.com/69636575/129479639-0669a94e-50d7-4fb3-b98e-c029c6084d82.png)


Just go to your github profile - settings - Developer settings - Personal access tokens - click on the name of the token you use for your avion repo (in my case it's named avion) 
![image](https://user-images.githubusercontent.com/69636575/129479698-9265c454-2153-4753-94d0-526c26535456.png)

Make sure the workflow tickbox is checked
![image](https://user-images.githubusercontent.com/69636575/129479738-8dc4f438-fa1d-46af-926d-9380e81c99d9.png)

Then click update found on the bottom of the page
![image](https://user-images.githubusercontent.com/69636575/129479756-8a020835-f8d1-4e5c-ac6a-51ae4de2da67.png)

**Reply by lea-tinoso** [reply]
I forgot to add: Please don't change the password on the Setup Database and Run Tests block inside the rspec yml. That'll cause issues on your Run Database and Run Tests block when you execute github action which was a pain in the a**
